extensions/resource-deployment/src/services/apiService.ts

```diff
@@ -5,26 +5,39 @@
 
 import * as azurecore from 'azurecore';
 import * as vscode from 'vscode';
+import * as arc from 'arc';
 
 export interface IApiService {
 	getAzurecoreApi(): Promise<azurecore.IExtension>;
+	getArcApi(): Promise<arc.IExtension>;
 }
 
 class ApiService implements IApiService {
 
 	private azurecoreApi: azurecore.IExtension | undefined;
+	private arcApi: arc.IExtension | undefined;
 
 	constructor() { }
 
 	public async getAzurecoreApi(): Promise<azurecore.IExtension> {
 		if (!this.azurecoreApi) {
-			this.azurecoreApi = <azurecore.IExtension>(await vscode.extensions.getExtension(azurecore.extension.name)?.activate());
+			this.azurecoreApi = <azurecore.IExtension>(await vscode.extensions.getExtension(azurecore.extension.name)?.exports);
 			if (!this.azurecoreApi) {
 				throw new Error('Unable to retrieve azurecore API');
 			}
 		}
 		return this.azurecoreApi;
 	}
+
+	public async getArcApi(): Promise<arc.IExtension> {
+		if (!this.arcApi) {
+			this.arcApi = <arc.IExtension>(await vscode.extensions.getExtension(arc.extension.name)?.exports);
```

Comment:
The await is unnecessary here now.

And really I'd suggest just getting rid of this whole function. Calling getExtension is only a small hit and will then guarantee nothing weird happens with a stale api object.

Reply:
keeping the functions as a shortcut to the longish getExtension ... exports invocation but simplifying it a lot.